pages/post/[id].tsx

```diff
@@ -0,0 +1,78 @@
+import { Layout } from 'components/Layout';
+import { Meta } from 'components/Meta';
+import { useRouter } from 'next/router';
+
+import React, { useState, useRef } from 'react';
+import { usePdf } from '@mikecousins/react-pdf';
+import { News } from 'components/Content/News';
+import Link from 'next/link';
+import { NewsSocials } from 'components/NewsSocials';
+import { ArrowLeft, ArrowRight } from 'lucide-react';
+
+const PdfContent = ({ id }: { id: string | string[] }) => {
+  const [page, setPage] = useState(1);
+  const canvasRef = useRef(null);
+
+  const { pdfDocument } = usePdf({
+    file: `/news-posts/${id}.pdf`,
+    page,
+    canvasRef,
+  });
+
+  return (
+    <div className="w-full py-4 sm:px-4 lg:p-[50px] border-y sm:border sm:border-gray-300">
+      {!pdfDocument && <span>Loading...</span>}
+      <canvas ref={canvasRef} className="w-full" />
+      {pdfDocument && pdfDocument.numPages && (
+        <div className="flex justify-end gap-4 mt-4 mr-4">
+          <button
+            disabled={page === 1}
+            onClick={() => setPage((prevPage) => prevPage - 1)}
+          >
+            <ArrowLeft
+              className={`max-md:w-4 ${page === 1 ? 'opacity-50' : ''}`}
+            />
+          </button>
+          <button
+            disabled={page >= pdfDocument.numPages}
+            onClick={() => setPage((prevPage) => prevPage + 1)}
+          >
+            <ArrowRight
+              className={`max-md:w-4 ${page >= pdfDocument.numPages ? 'opacity-50' : ''}`}
+            />
+          </button>
+        </div>
+      )}
+      <NewsSocials />
+    </div>
+  );
+};
+
+const Post = () => {
+  const router = useRouter();
+  const { id } = router.query;
+
+  if (!id) {
+    return <div>Loading...</div>;
+  }
+
+  return (
+    <Layout>
+      <Meta />
```

Comment:
wonder if we could have title/description here in meta data, I think you can get it from data/posts by id

Reply:
or can figure out all meta tags separately where it's missed